### ISSUE TITLE
🐛 [BUG] App Store GUI reports apps installed by mas as outdated but mas doesn't

### ISSUE THREAD

**Issue by rgoldberg** [mas-cli/mas#387]
## Your Environment

- `mas version`: 1.8.2
- macOS version (`system_profiler SPSoftwareDataType -detailLevel mini`): 11.3.1 (20E241)

### mas Install Method

- [X] `brew install mas` (homebrew-core)
- [] [mas-cli/tap](https://github.com/mas-cli/homebrew-tap)
- [] `.pkg` installer from [releases](https://github.com/mas-cli/mas/releases)
- [] Built from source
  - Fork/branch: ? (e.g. mas-cli/master)
  - Xcode version: 10.?

## Describe the Bug

The App Store GUI reports both Keynote (ID 409183694) & Apple Developer (ID 640199958) as outdated in both the Updates tab & in the Purchased tab of my account tab.

mas doesn't report them as outdated.

The App Store reports the new versions of Keynote as 11.0.1 & Developer as 9.0.1.

I already have these 2 versions installed: 11.0.1 (7030.1.2) & 9.0.1 (901.1.2), respectively.

Either:

- the App Store is incorrectly reporting outdated versions in 2 places
- the App Store somehow knows that the existing installations of these 2 apps must be reinstalled because they are corrupted
- Apple released new builds of these 2 apps with the same respective version numbers, which is causing mas to not recognize them as outdated

This has been going on for a few months, always with the then-latest stable mas & macOS…

## To Reproduce

1. Run `mas outdated` when Keynote 11.0.1 (7030.1.2) & Developer 9.0.1 (901.1.2) are installed.  Maybe this problem won't occur on others' computers…

**Comment by rgoldberg** [comment]
Keynote 11.1 & Developer 9.2 were recently released.  `mas upgrade` upgraded both according to its output & to `mas list`, but the App Store GUI says that an upgrade for Developer to 9.0.1 is available.  When I run Developer, its version is 9.2 (920.34.2).  The App Store GUI shows that 9.2 was released yesterday, but is offering to upgrade to 9.0.1.  Very odd…

**Comment by chris-araman** [comment]
This sounds like a bug in the App Store app, not in `mas`. Unfortunately, I don't think there's much we can do about it on our end. We don't work for Apple.

The next time you see this, please try selecting "Reload Page" from the "Store" menu when the Updates panel is visible.

**Comment by rgoldberg** [comment]
Yeah.  Seemed like an App Store issue, but wasn't sure that mas didn't somehow mess up app installs that somehow caused problems with the App Store.  Figured it was better to document this than to just ignore it, even if there isn't really much to be done from the mas side right now.

**Comment by rgoldberg** [comment]
@chris-araman

A few days ago, the App Store GUI reported a Developer 9.2.1 update as being available.

I used `mas upgrade` to install that update on the 4th.

The App Store GUI then no longer showed Developer as needing to be updated.

Today (the 6th) the App Store GUI now shows a Developer 9.2.1 update as being available, again.

It shows that it was released 3 days ago (the 3rd).

Menu > `Developer` > `About Developer` says it's version 9.2.1 (921.2.2).

I'm not sure if I checked the App Store GUI between the 4th & today, but I know I checked it right after mas installed Developer 9.2.1 to ensure that it was no longer listed in the GUI.

**Comment by chris-araman** [comment]
@rgoldberg, did the "Reload Page" steps I mention have any effect?

It's possible that Apple released Developer 9.2.1 twice. There have been several updates this week leading up to WWDC.

**Comment by rgoldberg** [comment]
"Reload Page" didn't do anything.

**Comment by rgoldberg** [comment]
`mas upgrade` upgraded Slack to 4.17.0 & Developer to 9.2.2, but the App Store GUI still thinks they can be upgraded to 4.17.0 & 9.2.2, respectively.

The App Store has issues recognizing app upgrades applied by `mas upgrade`.

I'm running the latest everything: macOS 11.4, mas 1.8.2, etc.

**Comment by orbanbalage** [comment]
I'm on Catalina 10.15.7 (19H1323), mas 1.8.2, and have the same issue.

Tried logging out and in, reloading the Update page, but no change.

I have four updates waiting in the AppStore, mas says there is nothing.

**Comment by chris-araman** [comment]
I've summarized this in the [Known Issues](https://github.com/mas-cli/mas#%EF%B8%8F-known-issues) section of the README.

**Comment by rgoldberg** [comment]
Possibly related:

The following versions of Prime Video are reported by their respective associated lookup URL.

```
10.47   https://itunes.apple.com/lookup?id=545519333&entity=desktopSoftware&country=US
10.48   https://itunes.apple.com/lookup?id=545519333&entity=desktopSoftware
10.48   https://itunes.apple.com/lookup?id=545519333
10.48   https://itunes.apple.com/lookup?id=545519333&country=US
```

We might just want to try to update every installed or specified app instead of checking first if each is outdated.
